File: Quiz_1/DeveshSrivastava_590017127/Quiz1DebuggingQuestion2.py
class Solution:

    def findMax(self, arr: list[int]) -> int:
        if not arr:
            return 0
        max_value = arr[0]
        for i in range(1, len(arr)):
            if arr[i] > max_value:
                max_value = arr[i]
        return max_value


# Problems with the original function:
# 1. Max variable is uninitialised
# 2. Uninitialised max value compared with elements in the array can cause undefined behavior


def main():
    solution = Solution()
    arr = [3, 1, 4, 1, 5, 9, 2, 6]
    print(arr)
    print(solution.findMax(arr))

File: Quiz_1/Raghav_Singhal_590012854/Quiz_1_Debugging_Question_2.c
#include<stdio.h>
int findMax(int arr[], int n) 
{
    int max=arr[0];
    for (int i = 1; i < n; i++) 
    {
        if (arr[i] > max) 
       {
            max = arr[i];
        }
    }
    return max;
}
void main()
{
    int a[]={1,2,3,4,5};
    printf("%d",findMax(a,5));
}